src/config/models.rs

```diff
@@ -25,6 +26,27 @@ pub struct ConfigOptsBuild {
     /// The public URL from which assets are to be served [default: /]
     #[structopt(long, parse(from_str=parse_public_url))]
     pub public_url: Option<String>,
+    /// Can be read only from config file
+    /// Optional pattern of the script to be injected instead of standard [default: None]
+    /// Can include {base}, {wasm}, {js}
+    #[structopt(skip)]
+    #[serde(default)]
+    pub pattern_script: Option<String>,
+    /// Can be read only from config file
+    /// Optional pattern of the preload links to be injected instead of standard [default: None]
+    /// Can include {base}, {wasm}, {js}
+    #[structopt(skip)]
+    #[serde(default)]
+    pub pattern_preload: Option<String>,
+    #[structopt(skip)]
+    #[serde(default)]
+    /// Can be read only from config file
+    /// Optional parameters of replacements inside
+    /// While {var} is being replaced with the provided value,
+    /// {@path} is replaced with contents of the provided file.
+    /// This allows insertion of some big JSON state or even HTML files
+    /// as a part of the `index.html` build
```

Comment:
```suggestion
    /// Optional replacement parameters corresponding to the patterns provided in
    /// `pattern_script` and `pattern_preload`.
    ///
    /// When a pattern is being replaced with its corresponding value from this map, if the value is
    /// prefixed with the symbol `@`, then the value is expected to be a file path, and the pattern
    /// will be replaced with the contents of the target file. This allows insertion of some big JSON
    /// state or even HTML files as a part of the `index.html` build.
    ///
    /// Trunk will automatically insert the `base`, `wasm` and `js` key/values into this map. In order
    //// for the app to be loaded properly, the patterns `{base}`, `{wasm}` and `{js}` should be used
    /// in `pattern_script` and `pattern_preload`.
    ///
    /// These values can only be provided via config file.
```

This suggested change is related to my comments below on the `pattern_evaluate` function.